routes/artifacts.js

```diff
@@ -571,44 +568,45 @@ api.declare({
 api.declare({
   method:     'get',
   route:      '/task/:taskId/runs/:runId/artifacts',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
   name:       'listArtifacts',
   stability:  base.API.stability.experimental,
   output:     'list-artifacts-response.json#',
   title:      "Get Artifacts from Run",
   description: [
-    "Returns a list of artifacts and associated meta-data for a given run."
+    "Returns a list of artifacts and associated meta-data for a given run.",
+    "",
+    "As a task may have many artifacts paging may be necessary. If this",
+    "end-point returns a `continuationToken`, you should call the end-point",
+    "again with the `continuationToken` as the query-string option:",
+    "`continuationToken`.",
+    "",
+    "By default this end-point will list up-to 1000 artifacts in a single page",
+    "you may limit this with the query-string parameter `limit`.",
   ].join('\n')
 }, async function(req ,res) {
-  var taskId = req.params.taskId;
-  var runId  = parseInt(req.params.runId);
+  let taskId        = req.params.taskId;
+  let runId         = parseInt(req.params.runId);
+  let continuation  = req.query.continuationToken || null;
+  let limit         = parseInt(req.query.limit || 1000);
   // TODO: Add support querying using prefix
 
```

Comment:
I was just mainly curious why we were checking it elsewhere but not here.  Nothing changed in this PR in regards to that so if we wanted to leave it unfixed or fix it in another PR at another time (just enter a bug?) then that's cool too.